diff --git a/include/config_mender_defines.h b/include/config_mender_defines.h
index 5bcfdea1f0..2dd7993f41 100644
--- a/include/config_mender_defines.h
+++ b/include/config_mender_defines.h
@@ -12,14 +12,14 @@
 #define MENDER_ROOTFS_PART_B_NUMBER 3
 #define MENDER_ROOTFS_PART_B_NUMBER_HEX 3
 #define MENDER_UBOOT_STORAGE_INTERFACE "mmc"
-#define MENDER_UBOOT_STORAGE_DEVICE 1
+#define MENDER_UBOOT_STORAGE_DEVICE 0
 
 /* BB variables. */
-#define MENDER_STORAGE_DEVICE_BASE "/dev/mmcblk1p"
+#define MENDER_STORAGE_DEVICE_BASE "/dev/mmcblk2p"
 #define MENDER_UBOOT_ENV_STORAGE_DEVICE_OFFSET_1 0x400000
 #define MENDER_UBOOT_ENV_STORAGE_DEVICE_OFFSET_2 0x600000
-#define MENDER_ROOTFS_PART_A_NAME "/dev/mmcblk1p2"
-#define MENDER_ROOTFS_PART_B_NAME "/dev/mmcblk1p3"
+#define MENDER_ROOTFS_PART_A_NAME "/dev/mmcblk2p2"
+#define MENDER_ROOTFS_PART_B_NAME "/dev/mmcblk2p3"
 
 /* For sanity checks. */
 #define MENDER_BOOTENV_SIZE 0x8000
